src/main/java/cubicchunks/world/column/CubeMap.java

```diff
@@ -208,6 +204,6 @@ private int binarySearch(int cubeY) {
 			}
 		}
 
-		return mid; // not found :(
+		return start; // not found :(
```

Comment:
one possible hack would he to have cube wrapper but right, in this case just leave the buiary search as is.